src/coll/mod.rs

```diff
@@ -414,23 +439,27 @@ impl Collection {
         self.client().execute_operation(op).await
     }
 
-    /// Inserts the documents in `docs` into the collection.
+    /// Inserts the data in `docs` into the collection.
     ///
     /// This operation will retry once upon failure if the connection and encountered error support
     /// retryability. See the documentation
     /// [here](https://docs.mongodb.com/manual/core/retryable-writes/) for more information on
     /// retryable writes.
     pub async fn insert_many(
         &self,
-        docs: impl IntoIterator<Item = Document>,
+        docs: impl IntoIterator<Item = T>,
         options: impl Into<Option<InsertManyOptions>>,
     ) -> Result<InsertManyResult> {
+        let mut translated_docs = Vec::new();
+        for doc in docs {
```

Comment:
Usually the idiomatic way to to this is to map to `Iterator<Item = Result<T>>` and then collect into `Result<Vec<T>>`. Since `Result` implements FromIterator as well as `Vec`, `collect` (which uses `FromIterator` under the hood) is able to collect into a `Result`. See [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6f913bd68ce6833f2f7f0b4cb46549cc) for an example.